### ISSUE TITLE
How to get daily data

### ISSUE THREAD

**Issue by QuocPhamBao** [metno/esd#266]
Hello
How can I modify this line to get the daily data from NCEP?
t2m <- t2m.NCEP(lon=c(0,360),lat=c(35,90))
If there is no way to get the daily data from NCEP, can we get daily data from ERA5 or some other datasets?

Best,

**Comment by brasmus** [comment]
Hi. The data that come with esd are only sample data and typically monthly or annual (to keep its size down). But you can download daily data with `ERA5.CDS()` after you have registered and installed the right python tools, and then read the the daily data using `retrieve`. I think you also can download daily data from e.g. http://apdrc.soest.hawaii.edu/datadoc/ncep2_daily.php and then read it using `retrieve`.

**Comment by brasmus** [comment]
Here is the way to extract only one day: ```r2
rr <- retrieve('hgt.1966.nc', lev = 500)
rr1 <- subset(rr, it = c("1966-01-08", "1966-01-08"))
rr2 <- subset(rr, it = c("1966-01-10", "1966-01-10"))
rr3 <- subset(rr, it = c("1966-01-13", "1966-01-13"))
rr123 <- c(rr1,rr2,rr3)
```

**Comment by kajsamp** [comment]
I'm guessing that the index of your object 'rr' is of a different class than your input 'it'. Could you check the following, please?

class(index(rr))
index(rr)[1:10]

**Comment by kajsamp** [comment]
The problem comes from the way you combine x1 and x2. Try using cbind(x1, x2) instead.

**Comment by QuocPhamBao** [comment]
Yes, I also tried that way, but there is another error:

Error in svd(y, nu = min(c(ny, npca)), nv = min(c(ny, npca))) : 
  a dimension is zero
Error in svd(t(y), nu = min(c(ny, npca)), nv = min(c(ny, npca))) : 
  a dimension is zero
Error: $ operator is invalid for atomic vectors

**Comment by kajsamp** [comment]
Sorry, it should be rbind(x1, x2), not cbind. 

You may still have trouble with the EOF analysis with such a small input object, containing only 8 time steps. What are you trying to figure out with the EOF analysis?

**Comment by QuocPhamBao** [comment]
Thank you !
May I ask one more question.
In the example 2.6, I think it will plot the figure of PC1 (first principle component), could we plot the figures of other PC, let's say PC2, PC3,PC4 and PC5?
Example 2.6.
#Get NCEP 2m air temperature for the selected spatial window defined by lon and lat
t2m <- t2m.NCEP(lon=c(-30,30),lat=c(40,70))
# Computes the EOFs
X <- EOF(t2m)
# Plot the result
plot(X)

**Comment by brasmus** [comment]
Sure! `plot(X,ip=2)` to show the 2nd component (`ip=3` shows the third, etc). We use a standard notation for arguments, where `is` refers to 'index space'. `it` is 'index time', and `ip` is index pattern.

**Comment by QuocPhamBao** [comment]
Thank you very much for your answer.
It helped me a lot.
Currently, can we create maps of EOFs in azimuthal projection ?

**Comment by QuocPhamBao** [comment]
Hello !
I would like to ask if possible to extract the values of 20 leading EOFs to see its values?
For the figure of ght (please see the attachment), is it possible to extract the data (longitude, latitude and its values) in that figure in order to draw in other application?
Best regards
![Rplot1](https://user-images.githubusercontent.com/26846020/193765684-4b3c96cf-93ff-4aa2-967b-966ac074e547.jpeg)

**Comment by brasmus** [comment]
Yes, you see the values of the EOFs by addressing them `attr(x,'pattern')` if `x <- EOF(...)`.  Alternatively, if you want just one pattern, you can ```r2
data("slp.ERA5")
eof <- EOF(slp.ERA5)
m1 <- subset(eof,ip=1)
map(m1) -> y
y
```

**Comment by QuocPhamBao** [comment]
Thank you for great support. I understand how to extract the value of eof.

In the below figure, is there any way for us to know how many percentage that component 1 accounts for, component 2 accounts for, .....component 20 accounts for.
I mean if I want to take the values and plot in another software, how I can see the values.

![20LeadingComPonents](https://user-images.githubusercontent.com/26846020/194851284-8273bd08-d861-4854-a036-4c1210d81440.jpg)

Best
